lib/nindo/feeds/sources.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Nindo.Sources do
+  @moduledoc false
+
+  alias NinDB.{Source}
+  alias Nindo.{Accounts, Post}
+
+  import Nindo.Core
+
+  def add(source, user) do
+    sources = user.sources
+    if source not in sources do
+      Accounts.change(:sources, sources ++ [source], user)
+    end
+  end
+
+  def remove(source, user) do
+    sources = user.sources
+    if source in sources do
+      Accounts.change(:sources, sources -- [source], user)
+    end
+  end
+
+  def generate(title, type, url) do
+    %Source{
+      title: title,
+      id: :erlang.phash2(url),
+      feed: url,
+      type: type,
+      icon: "https://#{url}/favicon.ico"
+    }
+  end
+
+  # we're only generating and caching the first five posts due to performance issues
+  def generate_posts(parsed_feed, source) do
+    parsed_feed["items"]
+    |> Enum.take(5) # remove to get entire feed
+    |> Enum.map(&Task.async(generate_post(&1, parsed_feed["title"], source)))
+    |> Task.await_many(30000)
+  end
+
+  def generate_post(entry, author, source) do
```

Comment:
Move this to `Nindo.Post`

Reply:
I'd rather keep it here. It generates a list of posts for a parsed feed, but for a specific source, so I think it belongs in sources.